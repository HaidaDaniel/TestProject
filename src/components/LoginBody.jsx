/** @format */
import { Row } from 'react-bootstrap'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import * as Yup from 'yup'

import {
  StyledFormContainer,
  StyledCol,
  StyledHeading,
  StyledFormGroup,
  StyledFormLabel,
  StyledFormControl,
  StyledSubmitButton,
} from '../styled/LoginBodyStyles'

const validationSchema = Yup.object().shape({
  email: Yup.string()
    .email('Please enter a valid email address')
    .required('This field is required'),
  password: Yup.string().required('This field is required'),
})

function LoginBody() {
  return (
    <Row>
      <StyledFormContainer>
        <StyledCol xs={12} md={6}>
          <StyledHeading>Login</StyledHeading>
          <Formik
            initialValues={{
              email: '',
              password: '',
            }}
            validationSchema={validationSchema}
            onSubmit={(values) => {
              console.log(values)
            }}>
            <Form>
              <StyledFormGroup>
                <StyledFormLabel>Email address</StyledFormLabel>
                <Field
                  name='email'
                  as={StyledFormControl}
                  type='email'
                  placeholder='Email address'
                />
                <ErrorMessage name='email' component='div' />
              </StyledFormGroup>
              <StyledFormGroup>
                <StyledFormLabel>Password</StyledFormLabel>
                <Field
                  name='password'
                  as={StyledFormControl}
                  type='password'
                  placeholder='Password'
                />
                <ErrorMessage name='password' component='div' />
              </StyledFormGroup>
              <StyledSubmitButton variant='primary' type='submit'>
                Login
              </StyledSubmitButton>
            </Form>
          </Formik>
        </StyledCol>
      </StyledFormContainer>
    </Row>
  )
}

export default LoginBody
