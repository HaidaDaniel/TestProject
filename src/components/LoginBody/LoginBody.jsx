/** @format */

import { Form, Button, Col, Row } from 'react-bootstrap'

function LoginBody() {
  return (
    <Row>
      <Form className='pt-3'>
        <Col xs={12} md={6} className='mx-auto border rounded pt-1'>
          <h3>Login</h3>
          <Form.Group className='my-3 mx-4' controlId='formGroupEmail'>
            <Form.Label>Email address or nickname</Form.Label>
            <Form.Control
              type='string'
              placeholder='Email address or nickname'
            />
          </Form.Group>
          <Form.Group className='mb-3 mx-4' controlId='formGroupPassword'>
            <Form.Label>Password</Form.Label>
            <Form.Control type='password' placeholder='Password' />
          </Form.Group>
          <Button variant='primary' type='submit' className='mb-3'>
            Login
          </Button>
        </Col>
      </Form>
    </Row>
  )
}
export default LoginBody
