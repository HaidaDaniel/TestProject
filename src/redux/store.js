import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './rootReducer';
import { authRootSaga, deAuthRootSaga } from './ducks/auth';
import { productRootSaga } from './ducks/product';
import { productsRootSaga } from './ducks/products';


const rootSagas = [authRootSaga, deAuthRootSaga, productsRootSaga, productRootSaga]

const sagaMiddleware = createSagaMiddleware();

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(sagaMiddleware)));

rootSagas.forEach((saga) => sagaMiddleware.run(saga));

export default store
    ;