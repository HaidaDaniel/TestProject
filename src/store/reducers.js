import { combineReducers } from 'redux';

const products = (state = [], action) => {
    switch (action.type) {
        case 'FETCH_PRODUCT_SUCCESS':
            return action.payload;
        default:
            return state;
    }
};

export default combineReducers({ products });