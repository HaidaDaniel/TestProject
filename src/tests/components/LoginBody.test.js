import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import LoginBody from '../../components/LoginBody';
<<<<<<< HEAD
<<<<<<< HEAD

=======
>>>>>>> origin/newTest
=======
>>>>>>> cdcb47e (1st refactor of tests description)
import wrapComponentInAllProviders from '../helpers/wrapComponentInAllProviders';
import store from '../../redux/store';

describe('LoginBody component', () => {
<<<<<<< HEAD
<<<<<<< HEAD
    it('should render login form', () => {
=======
    it('should renders login form', () => {
>>>>>>> origin/newTest
=======
    it('should renders login form', () => {
>>>>>>> cdcb47e (1st refactor of tests description)
        render(
            wrapComponentInAllProviders(<LoginBody />)
        );

        expect(screen.getByText('Email address')).toBeInTheDocument();
        expect(screen.getByText('Password')).toBeInTheDocument();
        expect(document.body).toMatchSnapshot();
    });

<<<<<<< HEAD
<<<<<<< HEAD
    it('should handle form submission', async () => {
=======
    it('should handles form submission', async () => {
>>>>>>> origin/newTest
=======
    it('should handles form submission', async () => {
>>>>>>> cdcb47e (1st refactor of tests description)


        render(
            wrapComponentInAllProviders(<LoginBody />))

        const emailInput = screen.getByPlaceholderText('Email address');
        const passwordInput = screen.getByPlaceholderText('Password');
        const loginButton = screen.getByRole('button');

        await (async () => {
            fireEvent.change(emailInput, { target: { value: 'demo@demo.com' } });
            fireEvent.change(passwordInput, { target: { value: 'password' } });
            fireEvent.click(loginButton);
            expect(document.body).toMatchSnapshot();
        });

        setTimeout(() => {
            const state = store.getState();
            expect(state.auth.user.email).toBe('demo@demo.com');
        }, 1000)
        clearTimeout();

    });
});
