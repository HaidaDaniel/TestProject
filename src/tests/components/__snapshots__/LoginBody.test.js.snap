// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<<< HEAD:src/tests/__snapshots__/components/LoginBody.snap.js
exports[`LoginBody component should render login form 1`] = `
========
exports[`LoginBody component should renders login form 1`] = `
>>>>>>>> origin/newTest:src/tests/components/__snapshots__/LoginBody.test.js.snap
<body>
  <div>
    <div
      class="row"
    >
      <div
        class="sc-bcPKhP hohMuK"
      >
        <div
          class="sc-grXZZQ dGlYIN col-md-6 col-12"
          data-testid="login col"
        >
          <h3
            class="sc-dkjKgF hKyBdL"
          >
            Login
          </h3>
          <form
            action="#"
          >
            <div
              class="sc-hLclGa jVqwjC"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Email address
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="email"
                placeholder="Email address"
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-hLclGa jVqwjC"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Password
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <button
              class="mb-3 btn btn-primary"
              data-testid="submit-button"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</body>
`;
